src/components/Form/Form.js

```diff
@@ -5,39 +5,51 @@ import { useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 
-export default function Form( { updateSalary, updateExpenses, updateZipcode} ) {
+export default function Form( { updateSalary, updateExpenses, updateZipcode, updateCombinedRate} ) {
 	const [salary, setSalary] = useState('');
   const [expenses, setExpenses] = useState('');
   const [zipcode, setZipcode] = useState('');
 	const navigate = useNavigate();
-
   function submitUserData(event) {
     event.preventDefault();
-		//do the setting state here
-		//after that's done, clearContent
-
-		updateSalary(salary);
-		updateExpenses(expenses);
-		updateZipcode(zipcode);
-
-		clearInput();
+		const validNumberRegex = /\d+/
+		const validZipcode = /^\d{5}(?:[-\s]\d{4})?$/gm;
 
-		// setTimeout(() => {
-      navigate('/result');
-    // }, 500);
+		if(!validNumberRegex.test(salary)){
```

Comment:
Seeing this duplicated code tells me there may be a way to refactor this so there isnt so much duplicated code - how can you DRY this up?